RevenueCatUI/Templates/Components/TemplateComponentsView.swift

```diff
@@ -4,28 +4,48 @@
 //
 //  Created by Josh Holtz on 6/11/24.
 //
+// swiftlint:disable missing_docs
 
 import RevenueCat
 import SwiftUI
 
 #if PAYWALL_COMPONENTS
 @available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
-struct TemplateComponentsView: View {
-
-    @Environment(\.locale)
-    var locale
+public struct TemplateComponentsView: View {
 
     let paywallComponentsData: PaywallComponentsData
+    let componentViewModels: [PaywallComponentViewModel]
 
-    init(paywallComponentsData: PaywallComponentsData) {
+    public init(paywallComponentsData: PaywallComponentsData, offering: Offering) {
         self.paywallComponentsData = paywallComponentsData
+
+        let components = paywallComponentsData.componentsConfig.components
+        self.componentViewModels = components.map { component in
+
+            // STEP 1 - Get list of preferred locales (and default)
+
+            // STEP 2 - Get localization for one of preferred locales in order
+            let chosenLocale = Locale.current
+            // TOOD: This logic is so wrong
```

Comment:
note: 
This comment is very ambiguous. It doesn't explain what's actually wrong with it, why, or how to fix it. 
And now it's a part of our `main` code, so anyone cloning the repo sees this. If I'm checking out this repo and find this, I run. 

In my first job, on my first day, I saw a comment that I'll never forget. It was something like:
```
// TODO: THIS IS VERY VERY WRONG. FIX BEFORE TRIP. 
```

And I checked the history and the comment had been there for > 7 years. Nobody knew what was actually wrong, so you can't really remove it. What trip? I'll never know. The author had left the company years before I joined.